src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,103 @@
 package core.basesyntax;
 
+import java.util.ArrayList;
+import java.util.List;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final double LOAD_FACTOR = 0.75;
+
+    private List<Entry<K, V>>[] buckets;
+    private int size;
+
+    public MyHashMap() {
+        buckets = new ArrayList[DEFAULT_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
+        if ((double) size / buckets.length > LOAD_FACTOR) {
+            resize();
+        }
 
+        int index = getIndex(key);
+        if (buckets[index] == null) {
+            buckets[index] = new ArrayList<>();
+        }
+        for (Entry<K, V> entry : buckets[index]) {
+            if (entry.getKey() == null) {
+                continue;
+            }
+            if (entry.getKey().equals(key)) {
+                entry.setValue(value);
+                return;
+            }
+        }
+        buckets[index].add(new Entry<>(key, value));
+        size++;
     }
 
     @Override
     public V getValue(K key) {
+        int index = getIndex(key);
+        if (buckets[index] != null) {
+            for (Entry<K, V> entry : buckets[index]) {
+                if (entry.getKey() == null && key == null) {
+                    return entry.getValue();
+                }
+                if (key.equals(entry.getKey())) {
+                    return entry.getValue();
+                }
+            }
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private int getIndex(K key) {
+        if (key == null) {
+            return 0;
+        }
+        return Math.abs(key.hashCode() % buckets.length);
```

Comment:
```suggestion
        return key == null ? 0 : Math.abs(key.hashCode() % buckets.length);
```
not a mistake but ternary looks nice here